File: VyaparAI.Api/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace VyaparAI.Api.Models;

public class Product
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = string.Empty;

    [BsonRepresentation(BsonType.ObjectId)]
    public string BusinessId { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;
    public string Sku { get; set; } = string.Empty;
    public string? Barcode { get; set; }
    public string Category { get; set; } = string.Empty;
    public string? Brand { get; set; }

    public decimal PurchasePrice { get; set; }
    public decimal SellingPrice { get; set; }
    public decimal GstPercentage { get; set; }

    public int CurrentQuantity { get; set; }
    public int MinimumStockLevel { get; set; }

    [BsonRepresentation(BsonType.ObjectId)]
    public string? SupplierId { get; set; }

    public DateTime? ExpiryDate { get; set; }
    public string Unit { get; set; } = "pcs";
    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: vyaparai-frontend/src/types/product.ts
import type { PagedQuery } from './common'

export type StockStatus = 'OK' | 'LOW' | 'OUT'

export interface Product {
  id: string
  name: string
  sku: string
  barcode?: string | null
  category: string
  brand?: string | null
  purchasePrice: number
  sellingPrice: number
  gstPercentage: number
  currentQuantity: number
  minimumStockLevel: number
  supplierId?: string | null
  supplierName?: string | null
  expiryDate?: string | null
  unit: string
  isActive: boolean
  stockStatus: StockStatus
  createdAt: string
  updatedAt: string
}

export interface ProductFormValues {
  name: string
  sku: string
  barcode?: string
  category: string
  brand?: string
  purchasePrice: number
  sellingPrice: number
  gstPercentage: number
  currentQuantity?: number // only sent on create
  minimumStockLevel: number
  supplierId?: string
  expiryDate?: string
  unit: string
  isActive?: boolean
}

export interface ProductQuery extends PagedQuery {
  category?: string
  lowStockOnly?: boolean
  includeInactive?: boolean
}
